File: Arrays/ContainsDuplicateII/Python/219-Contains-Duplicate-II.go
func containsNearbyDuplicate(nums []int, k int) bool {
    compMap := make(map[int]int)

    for i, num := range nums {
        if lastIdx, exists := compMap[num]; exists {
            distance := i - lastIdx
            if distance <= k {
                return true
            }
        }
        compMap[num] = i
    }

    return false    
    
}

File: Arrays/ContainsDuplicateII/Python/219-Contains-Duplicate-II.py
class Solution:
    def containsNearbyDuplicate(self, nums: List[int], k: int) -> bool:
        comp_map = {}

        for i,v in enumerate(nums): 
            if v in comp_map and abs(i-comp_map[v]) <= k:
                    return True
                
            comp_map[v] = i
            
        return False